wagtail/admin/views/editing_sessions.py

```diff
@@ -62,18 +63,73 @@ def ping(request, app_label, model_name, object_id, session_id):
+    # are merged, such that the most recently seen one is reported, but is_editing is true
+    # if any session has the editing flag set (not just the latest one).
+    other_sessions_lookup = {}
+    for other_session in other_sessions:
+        try:
+            other_session_info = other_sessions_lookup[other_session.user.pk]
+        except KeyError:
+            other_sessions_lookup[other_session.user.pk] = {
+                "session_id": other_session.id,
+                "user": other_session.user,
+                "last_seen_at": other_session.last_seen_at,
+                "is_editing": other_session.is_editing,
+                "revision_id": None,
+            }
+        else:
+            if other_session.is_editing:
+                other_session_info["is_editing"] = True
+
+    revision_id = request.GET.get("revision_id", None)
+    if revision_id is not None and issubclass(model, RevisionMixin):
+        all_revisions = obj.revisions.defer("content")
+        try:
+            original_revision = all_revisions.get(id=revision_id)
+        except Revision.DoesNotExist:
+            raise Http404
+
+        newer_revisions = (
+            all_revisions.filter(created_at__gt=original_revision.created_at)
+            .order_by("created_at")
+            .select_related("user")
+        )
+
+        for revision in newer_revisions:
+            try:
+                session_info = other_sessions_lookup[revision.user_id]
+            except KeyError:
+                other_sessions_lookup[revision.user_id] = {
+                    "session_id": None,
+                    "user": revision.user,
+                    "last_seen_at": revision.created_at,
+                    "is_editing": False,
+                    "revision_id": revision.id,
+                }
+            else:
+                session_info["revision_id"] = revision.id
+                if revision.created_at > session_info["last_seen_at"]:
+                    session_info["last_seen_at"] = revision.created_at
+
     return JsonResponse(
         {
             "session_id": session.id,
             "other_sessions": [
                 {
-                    "session_id": other_session.id,
-                    "user": get_user_display_name(other_session.user),
-                    "last_seen_at": other_session.last_seen_at.isoformat(),
+                    "session_id": other_session["session_id"],
+                    "user": get_user_display_name(other_session["user"]),
+                    "last_seen_at": other_session["last_seen_at"].isoformat(),
+                    "is_editing": other_session["is_editing"],
+                    "revision_id": other_session["revision_id"],
                 }
-                for other_session in other_sessions
+                for other_session in sorted(
+                    other_sessions_lookup.values(),
+                    key=lambda other_session: other_session["last_seen_at"],
+                )
```

Comment:
Now that I think about it again, I think the default sorting for the remaining ones should be sorted ascending by the ID, to avoid shifting the list too much.

If we sort ascending by `last_seen_at` the following would happen:
- User A joins the session and pings. `A.last_seen_at = 1`
  - Result: [A]
- User B joins the session and pings. `B.last_seen_at = 2`
  - Result: [A, B]
- User A pings. `A.last_seen_at = 3`
  - Result: [B, A]
- User B pings. `B.last_seen_at = 4`
  - Result: [A, B]
- Repeat.

If we sort descending, the same thing would eventually happen, just switch around.

If we sort ascending by the ID, any new user would be appended to the end, and the order would be mostly stable as long as they're still connected. Disconnecting and reconnecting would remove and re-append the session to the end of the list, which is fine I think.